pkg/kubelet/kubelet.go

```diff
@@ -2276,9 +2277,10 @@ func (kl *Kubelet) deletePod(pod *v1.Pod) error {
 func (kl *Kubelet) rejectPod(pod *v1.Pod, reason, message string) {
 	kl.recorder.Eventf(pod, v1.EventTypeWarning, reason, message)
 	kl.statusManager.SetPodStatus(pod, v1.PodStatus{
-		Phase:   v1.PodFailed,
-		Reason:  reason,
-		Message: "Pod was rejected: " + message})
+		QOSClass: v1qos.GetPodQOS(pod),
```

Comment:
is QoS the only field we are dropping here?

Is there a reason to have this field or you just trying to protect from the potential enforcement of the QoS class to be non-mutable?

Reply:
> you just trying to protect from the potential enforcement of the QoS class to be non-mutable?

Yes, it is.

@SergeyKanzhelev  As far as I know, it is the only missing field according to the current status structure. The `startTime` field will be filled in [updateStatusInternal](https://github.com/carlory/kubernetes/blob/fix-126662-kubelet/pkg/kubelet/status/status_manager.go#L603) which is called by `SetStatus` method. Other fields are empty until the pod is accepted by kubelet. I don't know whether other external programs will fill other fields, i.e. `status.conditions`.

If the pod's qosClass is forbidden to update, we can handle it like `startTime` in `updateStatusInternal`, for example,

```go
// ensure that the qos class does not change across updates.
status.QOSClass = v1qoshelper.GetPodQOS(pod)
```

Or if we worry about missing other fields, we can do it like the following in `rejectPod`,

```go
status := pod.Status.DeepCopy()
status.Phase = v1.PodFailed
status.Reason = reason
status.Message = "Pod was rejected: " + message
```

I'm not sure if this is intentional to clear other fields when`rejectPod` method is called.

Reply:
@tallclair once we will have the total request status that is calculated on pod creation, we will need to make sure we are not removing those fields on admission failure